includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -175,6 +179,24 @@ public function set_session() {
 		WC()->session->set_customer_session_cookie( true );
 	}
 
+	/**
+	 * Handles payment request redirect when the redirect dialog "Continue" button is clicked.
+	 */
+	public function handle_payment_request_redirect() {
+		if (
+			! empty( $_GET['wcpay_payment_request_redirect_url'] )
+			&& ! empty( $_GET['_wpnonce'] )
+			&& wp_verify_nonce( $_GET['_wpnonce'], 'wcpay-set-redirect-url' ) // @codingStandardsIgnoreLine
+		) {
+			$url = rawurldecode( esc_url_raw( wp_unslash( $_GET['wcpay_payment_request_redirect_url'] ) ) );
+			// Sets a redirect URL cookie for 10 minutes,
+			// which we will redirect to after authentication.
+			wc_setcookie( 'wcpay_payment_request_redirect_url', $url, time() + MINUTE_IN_SECONDS * 10 );
```

Comment:
What happens if the 10 minutes passes and the user logs in after that?

Reply:
If 10 minutes passes they're not redirected to the same page after login. This 10 minute cookie is an arbitrary timeout to login and complete the PRB transaction.